src/sounds.h

```diff
@@ -3,12 +3,31 @@
 
 #include "enums.h" // For point
 #include "cursesdef.h" // For WINDOW
+#ifdef SDL_SOUND
+#include "SDL2/SDL_mixer.h"
+#endif
 
 #include <vector>
 #include <string>
 
 class monster;
 class player;
+class Creature;
+struct sound_effect;
+
+struct sound_effect {
+    std::vector<std::string> files;
+    std::string id;
+    std::string variant;
+    int volume;
+    Mix_Chunk *chunk;
```

Comment:
`Mix_Chunk` will only be defined if "SDL_mixer.h" has been included. This totally breaks no-sound builds.

You should move the definition of this struct into the cpp file (and also remove `get_channel` below), then you don't need to include "SDL_mixer.h" in the header at all. (Still needed in the cpp file.)

Neither this struct, nor the declaration of `get_channel` needs to be in the header, they are not used outside of the one cpp file which also contains their actual implementation. In other words: no other cpp file needs to know about them.